### PR TITLE
feature that let you get a new picture from the db

### PR DESCRIPTION
**Pull request by BoostedBoat**
Added methods for the users that let you get a new picture from the db: if the user is signed in it only returns pictures that haven't already been guessed and pictures that the user didn't upload

### CODE REVIEW

**Comment by JeremyDezalos** [comment]
why did you requested me if you cancel it right away ?

**Comment by JeremyDezalos** [comment]
WHY DID YOU DELETED MY COMMMMETN ??!

File: app/src/main/java/com/github/wnder/GuestUser.java [hunk 1]
```diff
@@ -11,4 +18,31 @@ public String getName(){
...
+        //If no image fits, return empty string
+        if(0 == allIds.size()){
+            return new String();
+        }
+        //else, return randomly chosen string
+        else{
+            List<String> idList = new ArrayList<>();
+            idList.addAll(allIds);
+            Random random = new Random();
+            int index = random.nextInt(allIds.size());
```
Comment (JeremyDezalos) [inline app/src/main/java/com/github/wnder/GuestUser.java:44]: How do you make sure this is a new photo that the user never got before ? I know this is for a guest but is there no  possibility to use a cache ?

**Review by JeremyDezalos — APPROVED** [review]

File: app/src/androidTest/java/com/github/wnder/UserTesting.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    public void getNewPictureForSignedInUserWorks() throws ExecutionException, InterruptedException, TimeoutException {
+        SignedInUser u = new SignedInUser("testUser", Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag));
+        String pic = u.getNewPicture();
+        FirebaseFirestore storage = FirebaseFirestore.getInstance();
+
+        //Check that it is not in user's uploaded and guessed pictures
+        Set<String> upAdownPics = new HashSet<>();
+        Task<DocumentSnapshot> task = storage.collection("users").document("testUser").get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
+            @Override
+            public void onSuccess(DocumentSnapshot documentSnapshot) {
```
Comment (rkuenzi-epfl) [inline app/src/androidTest/java/com/github/wnder/UserTesting.java:45]: As the `addOnSuccessListener` make it clear what will be in the following function, you could use a lambda here:
```
addOnSuccessListener((DocumentSnapshot documentSnapshot) -> {
// onSuccess content goes here
});
```
(Same goes for all other onSuccessListener in the PR)

**Review by rkuenzi-epfl — APPROVED** [review]
Lambdas are nice

**Pull request merged**